src/crimson/os/seastore/cache.cc

```diff
@@ -1117,7 +1117,12 @@ record_t Cache::prepare_record(
       if (segment_provider != nullptr &&
           i->get_paddr().get_addr_type() == paddr_types_t::SEGMENT) {
         auto sid = i->get_paddr().as_seg_paddr().get_segment_id();
-        auto &sinfo = segment_provider->get_seg_info(sid);
+	auto provider = segment_provider;
+	if (cold_segment_provider &&
+	    !segment_provider->contains(sid)) {
+	  provider = cold_segment_provider;
```

Comment:
```c++
ceph_assert(provider->contains(sid));
```

Reply:
The comment is dismissed, see https://github.com/ceph/ceph/pull/47974#discussion_r1061095999